### DISCUSSION TITLE
编译OpenWRT如何更换内核小版本

### DISCUSSION THREAD

**Discussion by Sm1rkBoy** [coolsnowwolf/lede#9993] [Q&A / 快速问答]
lean大的源码现在是5.15大版本的内核,小版本是.60,Linux Kernel最新5.15版本小版本内核为.63,如何在编译时将.60小版本升级为.63小版本呢

**Comment by HongJie-Master** [comment]
lede/target/linux/x86/Makefile
把KERNEL_PATCHVER:=5.15 改成 5.4

**Reply by Sm1rkBoy** [reply]
这个是更换大版本，我想知道的是如何更换小版本5.15.60如何更换到5.15.63

**Comment by ZeaKyX** [comment]
只能根据 git 回退了，当然 patch 也得对应。

**Comment by gtcar2020** [comment]
在include文件夹下面的kernel-5.15文件里面改版本号和对应稀哈值

**Reply by worldmaomao** [reply]
我把这里的版本改为5.15.41，结果编译完之后，内核版本却是5.15.96.。难道编译的时候，会自动pull内核代码？

**Reply by gtcar2020** [reply]
改.config文件

**Comment by zfhxi** [comment]
总结上述大佬的回答，所以这样操作是对的吗？
```bash
git clone https://github.com/coolsnowwolf/lede
cd lede
# change kernel to 5.10.176
git reset --hard f4acd50
sed -i 's?KERNEL_PATCHVER:=5.4?KERNEL_PATCHVER:=5.10?g' target/linux/<架构>/Makefile
```
